x-pack/plugins/fleet/server/services/agents/request_diagnostics_action_runner.ts

```diff
@@ -46,18 +46,15 @@ export async function requestDiagnosticsBatch(
   const actionId = options.actionId ?? uuidv4();
   const total = options.total ?? givenAgents.length;
 
-  const supportedAgents: Agent[] = givenAgents.reduce((acc: Agent[], agent: Agent) => {
-    if (isAgentRequestDiagnosticsSupported(agent)) {
-      acc.push(agent);
-    } else {
+  givenAgents.forEach((agent: Agent) => {
+    if (!isAgentRequestDiagnosticsSupported(agent)) {
```

Comment:
if we keep the unsupported agent id in the action doc, the UI can report the failed status of the bulk diagnostics action:
<img width="1564" alt="image" src="https://user-images.githubusercontent.com/90178898/217559793-a8f0e813-7caa-48e2-b431-023330ca1fe0.png">